lib/elixir/lib/file.ex

```diff
@@ -1340,6 +1355,28 @@ defmodule File do
   defp normalize_modes([], true), do: [:binary]
   defp normalize_modes([], false), do: []
 
+  defp normalize_write_modes(modes) do
+    normalize_modes([:write] ++ modes, true)
+    |> set_raw_unless_encoding_specified
+  end
+
+  # Using raw mode means we get the performance benefit of using `writev`.
+  # However, Erlang will raise a badarg error if a file encoding is specified
+  # and raw mode is also requested.
+  # http://erlang.org/doc/man/file.html#open-2
+  defp set_raw_unless_encoding_specified(modes) do
+    case specifies_encoding?(modes) do
```

Comment:
Probably `List.keymember?/3` will better communicate what data structure we are working with here (even if it's totally the same technically as `Keyword.has_key?/2`). :bowtie: